src/heaps/binomial_heap.jl

```diff
@@ -0,0 +1,368 @@
+    iter1 = iterate(l1)
+    iter2 = iterate(l2)
+    while (iter1 !== nothing && iter2 !== nothing)
+        (elem1, state1) = iter1
+        (elem2, state2) = iter2
+        if( elem1.degree <= elem2.degree) 
+            push!(_new.rootList, elem1)
+            iter1 = iterate(l1, state1)
+        else
+            push!(_new.rootList, elem2)
+            iter2 = iterate(l2, state2)
+        end
+    end
+    while (iter1 !== nothing)
+        (elem1, state1) = iter1 
+        push!(_new.rootList, elem1) 
+        iter1 = iterate(l1, state1) 
+    end
+    while (iter2 !== nothing)
+        (elem2, state2) = iter2
+        push!(_new.rootList, elem2)
+        iter2 = iterate(l2, state2)
+    end
+    h1.rootList = _new.rootList
+    adjust(h1)
+    return 
+end
+
+function push!(_heap::MutableBinomialHeap{T}, key) where T
+    handle = _heap.cumm_nodecount + 1
+    temp = MutableBinomialHeapNode(convert(T, key), handle)
+    insertATreeInHeap(_heap, temp)
+    _heap.nodecount = _heap.nodecount + 1
+    _heap.cumm_nodecount = _heap.cumm_nodecount + 1
+    return temp.handle
+end
+#################################################
+#
+#   interfaces
+#
+#################################################
+length(h::MutableBinomialHeap) = h.nodecount
+
+isempty(h::MutableBinomialHeap) = isempty(h.rootList)
+
+"""
+union!(h1::MutableBinomialHeap, h2::MutableBinomialHeap)
+
+merges heap `h2` into heap `h1`
+"""
+union!(h1::MutableBinomialHeap, h2::MutableBinomialHeap) = union!(h1, h2, true)
+
+
+function sizehint!(h::MutableBinomialHeap, s::Integer)
+    sizehint!(h.rootList, s)
+    return h
+end
+
+@inline top(heap::MutableBinomialHeap) =  top_with_handle(heap)[1]
+
+minimum(heap::MutableBinomialHeap) =  top_with_handle(heap)[1]
+"""
+top_with_handle(h::MutableBinomialHeap)
+
+Returns the minimum element of the heap `h` and its handle.
+"""
+function top_with_handle(h::MutableBinomialHeap) 
+    (val, node) = top_elem_with_handle(h)
+    return (val, node.handle)
+end
+
+"""
+pop!(h::MutableBinomialHeap)
+
+Returns the minimum element of the heap `h` and its handle and alo deletes it from heap
+"""
+pop!(h::MutableBinomialHeap{T}) where {T} = extractMin!(h)
+popmin!(h::MutableBinomialHeap{T}) where {T} = extractMin!(h)
+"""
+update!{T}(h::MutableBinomialHeap{T}, i::Int, new_val::T)
+
+Replace the element having handle 'i' in heap `h` with `new_val`.
+"""
+update!(h::MutableBinomialHeap{T}, i::Int, new_val) where T = decreaseKey!(h, i, new_val)
+
+"""
+delete!{T}(h::MutableBinomialHeap{T}, i::Int)
+
+Deletes the element with handle 'i' from heap `h` .
+"""
+function delete!(heap::MutableBinomialHeap{T}, i::Int) where T 
+    n = length(heap.rootList)
+    if n == 0
+        return nothing
+    end
+    decreaseKey!(heap, i, typemin(T)) 
+    extractMin!(heap)
+    return heap
+end
+
+setindex!(h::MutableBinomialHeap, v, i::Int) = update!(h, i, v)
+getindex(h::MutableBinomialHeap, i::Int) =  findNode(h, i).data
```

Comment:
These two functions are not covered in the tests, but they should be.

Reply:
Yeah i add them